libs/librrgraph/src/utils/alloc_and_load_rr_indexed_data.cpp

```diff
@@ -638,20 +641,24 @@ static void load_rr_indexed_data_T_values(const RRGraphView& rr_graph,
  * It is not safe to assume that each node of the same wire type has the same switches with the same
  * delays, therefore we take their average to take into account the possible differences
  */
-static void calculate_average_switch(const RRGraphView& rr_graph, int inode, double& avg_switch_R, double& avg_switch_T, double& avg_switch_Cinternal, int& num_switches, short& buffered, vtr::vector<RRNodeId, std::vector<RREdgeId>>& fan_in_list) {
+static void calculate_average_switch(const RRGraphView& rr_graph, int inode, double& avg_switch_R, double& avg_switch_T, double& avg_switch_Cinternal, int& num_switches, int& num_shorts, short& buffered, vtr::vector<RRNodeId, std::vector<RREdgeId>>& fan_in_list) {
     auto node = RRNodeId(inode);
 
     avg_switch_R = 0;
     avg_switch_T = 0;
     avg_switch_Cinternal = 0;
     num_switches = 0;
+    num_shorts = 0;
     buffered = UNDEFINED;
     for (const auto& edge : fan_in_list[node]) {
         /* want to get C/R/Tdel/Cinternal of switches that connect this track segment to other track segments */
         if (rr_graph.node_type(node) == CHANX || rr_graph.node_type(node) == CHANY) {
             int switch_index = rr_graph.rr_nodes().edge_switch(edge);
 
-            if (rr_graph.rr_switch_inf(RRSwitchId(switch_index)).type() == SwitchType::SHORT) continue;
+            if (rr_graph.rr_switch_inf(RRSwitchId(switch_index)).type() == SwitchType::SHORT) {
+                num_shorts++;
+                continue;
+            }
```

Comment:
With this code change, it seems I could get divide by 0 errors late in the routine if a wire is only driven by a short (e.g. it might be the second part of an L-shaped wire).  CI seems to have errors, so I think there's a bug here.